src/language-server/logger.ts

```diff
@@ -19,9 +19,20 @@ import Settings from './settings';
 
 export class Logger {
   private logFile: WriteStream|undefined = undefined;
+  private readonly logFileFromFlag: WriteStream|undefined;
   private settings: Settings|undefined;
   private queue: string[] = [];
-  constructor(private readonly connection: IConnection) {
+  constructor(
+      private readonly connection: IConnection,
+      logToFileFlag: string|undefined) {
```

Comment:
I would make this an object with default value `{}`. That makes it more readable on the call-side, as in the test you had to add `undefined` as second argument which is ambigous. `new Logger(connection)` and `new Logger(connection, {logToFileFlag: true})` is better imo.

Reply:
For internal types I care much more about correctness than the convenience of optional properties and args, but the legibility gains of named args are nice here.